onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -2370,6 +2391,18 @@ def aten_equal(self: TTensor, other: TTensor) -> BOOL:
     return op.Equal(sum_of_abs, 0)
 
 
+@torch_op("aten::equal")
+def aten_equal_bool(self: BOOL, other: BOOL) -> BOOL:
```

Comment:
why can't we directly onnx Equal self and other? Please also comment it thx.

Reply:
I don't know what I was thinking. For some reason I thought equal didn't work on bools. Updated.

Reply:
Probably because how aten_equal was implemented up there? It is also not directly onnx equal. This question applies there as well. I assume there is something I'm missing.

Reply:
Hmm what's going on. Let me try to merge that.

Reply:
Done. Thanks for catching this!

Reply:
Okay, I realized the torch.equal is probably returning single True/False for the result, while ONNX one does elementwise comparison. Let's add the comment ..

Reply:
Added